shell/src/test/java/org/apache/accumulo/shell/commands/SetIterCommandTest.java

```diff
@@ -0,0 +1,114 @@
+ * limitations under the License.
+ */
+package org.apache.accumulo.shell.commands;
+
+import java.io.PrintWriter;
+import java.io.Writer;
+import java.lang.reflect.Field;
+import java.util.EnumSet;
+
+import org.apache.accumulo.core.client.Connector;
+import org.apache.accumulo.core.client.IteratorSetting;
+import org.apache.accumulo.core.client.admin.TableOperations;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.shell.Shell;
+import org.apache.accumulo.start.classloader.vfs.AccumuloVFSClassLoader;
+import org.apache.commons.cli.CommandLine;
+import org.easymock.EasyMock;
+import org.junit.Before;
+import org.junit.Test;
+
+import jline.console.ConsoleReader;
+
+public class SetIterCommandTest {
+  private SetIterCommand cmd;
+
+  @Before
+  public void setup() {
+    cmd = new SetIterCommand();
+
+    // Initialize that internal state
+    cmd.getOptions();
+  }
+
+  @Test
+  public void addColumnAgeOffFilter() throws Exception {
+    Connector conn = EasyMock.createMock(Connector.class);
+    CommandLine cli = EasyMock.createMock(CommandLine.class);
+    Shell shellState = EasyMock.createMock(Shell.class);
+    ConsoleReader reader = EasyMock.createMock(ConsoleReader.class);
+    Writer out = EasyMock.createMock(PrintWriter.class);
+    TableOperations tableOperations = EasyMock.createMock(TableOperations.class);
+
+    // Command line parsing
+    EasyMock.expect(cli.hasOption("all")).andReturn(true);
+    EasyMock.expect(cli.hasOption("all")).andReturn(true);
+    EasyMock.expect(cli.hasOption("all")).andReturn(true);
+    EasyMock.expect(cli.hasOption("t")).andReturn(true);
+    EasyMock.expect(cli.hasOption("t")).andReturn(true);
+    EasyMock.expect(cli.getOptionValue("t")).andReturn("foo");
+    EasyMock.expect(cli.hasOption("ns")).andReturn(false);
+    EasyMock.expect(cli.getOptionValue("p")).andReturn("21");
+    EasyMock.expect(cli.getOptionValue("class")).andReturn("org.apache.accumulo.core.iterators.user.ColumnAgeOffFilter");
+    EasyMock.expect(cli.hasOption("ageoff")).andReturn(false);
+    EasyMock.expect(cli.hasOption("agg")).andReturn(false);
+    EasyMock.expect(cli.hasOption("regex")).andReturn(false);
+    EasyMock.expect(cli.hasOption("reqvis")).andReturn(false);
+    EasyMock.expect(cli.hasOption("vers")).andReturn(false);
+    EasyMock.expect(cli.getOptionValue("n", null)).andReturn(null);
+
+    // Loading the class
+    EasyMock.expect(shellState.getClassLoader(cli, shellState)).andReturn(AccumuloVFSClassLoader.getClassLoader());
+
+    // Set the output object
+    Field field = reader.getClass().getSuperclass().getDeclaredField("out");
+    field.setAccessible(true);
+    field.set(reader, out);
+
+    // Parsing iterator options
+    reader.flush();
+    EasyMock.expectLastCall().times(3);
+
+    EasyMock.expect(shellState.getReader()).andReturn(reader);
+    EasyMock.expect(reader.readLine(EasyMock.anyObject(String.class))).andReturn(""); // Default of 'false'
```

Comment:
Is this when the shell ask for the negate option?  If so would be nice to comment for each of these line what the shell is asking for.

Reply:
Yup -- pushed some comments.